pkg/cmd/roachprod-microbench/github.go

```diff
@@ -43,6 +43,47 @@ func createBenchmarkPostRequest(
 	return formatter, req
 }
 
+// regressionInfo holds information about a single benchmark regression.
+type regressionInfo struct {
+	benchmarkName  string
+	metricUnit     string
+	percentChange  float64
+	formattedDelta string
+}
+
+// createRegressionPostRequest creates a post request for benchmark performance regressions.
+func createRegressionPostRequest(
+	pkgName string, regressions []regressionInfo, sheetLink, description string,
+) (issues.IssueFormatter, issues.PostRequest) {
+	// Build the regression summary message
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("Performance regressions detected in package %s\n\n", pkgName))
+	sb.WriteString(fmt.Sprintf("Comparison: %s\n\n", description))
+	sb.WriteString(fmt.Sprintf("Found %d benchmark(s) with regressions ≥%.0f%%:\n\n", len(regressions), slackPercentageThreshold))
+
+	for i, reg := range regressions {
+		if i >= 10 { // Limit to 10 regressions in the summary
+			sb.WriteString(fmt.Sprintf("... and %d more regression(s)\n", len(regressions)-i))
+			break
+		}
+		sb.WriteString(fmt.Sprintf("• %s (%s): %s (%.1f%%)\n",
+			reg.benchmarkName, reg.metricUnit, reg.formattedDelta, reg.percentChange))
+	}
+
+	sb.WriteString(fmt.Sprintf("\nDetailed comparison: %s\n", sheetLink))
+
+	benchmarkName := regressions[0].benchmarkName
```

Comment:
See my comment about finding the correct owner [1].

[1] https://github.com/cockroachdb/cockroach/pull/157045#issuecomment-3649977842